Python/optimizer_analysis.c

```diff
@@ -1064,76 +880,52 @@ uop_abstract_interpret_single_inst(
         }
 
         case _INIT_CALL_PY_EXACT_ARGS: {
-            _Py_UOpsSymbolicValue **__args_;
-            _Py_UOpsSymbolicValue *__self_or_null_;
-            _Py_UOpsSymbolicValue *__callable_;
-            _Py_UOpsSymbolicValue *__new_frame_;
-            __args_ = &stack_pointer[-oparg];
-            __self_or_null_ = stack_pointer[-1 - oparg];
-            __callable_ = stack_pointer[-2 - oparg];
-            // Store the frame symbolic to extract information later
-            assert(ctx->new_frame_sym.func == NULL);
-            ctx->new_frame_sym.func = __callable_;
-            ctx->new_frame_sym.self_or_null = __self_or_null_;
-            ctx->new_frame_sym.args = __args_;
-            __new_frame_ = _Py_UOpsSymbolicValue_New(ctx, NULL);
-            if (__new_frame_ == NULL) {
-                goto error;
-            }
-            stack_pointer[-2 - oparg] = (_Py_UOpsSymbolicValue *)__new_frame_;
+            // Don't put in the new frame. Leave it be so that _PUSH_FRAME
+            // can extract callable, self_or_null and args later.
+            // Set stack pointer to the callable.
             stack_pointer += -1 - oparg;
```

Comment:
Oh wait -- I didn't realize that you bumped the args off the stack already here! This feels too drastic. We shouldn't rely on there being nothing between `_INIT_CALL_PY_EXACT_ARGS` and `_PUSH_FRAME` that touches the stack. The latter can decrement the stack pointer, it has the arg count in `oparg`.

(The "real" VM certainly leaves the args on the stack here.)